### ISSUE TITLE
A11y question: table footnotes

### ISSUE THREAD

**Issue by bsouster** [wet-boew/wet-boew#7485]
It's common practice for table footnotes to use asterisks (*, **, ***, etc). 
When there are multiple tables on a page, sometimes this results in several links with the same link text linking to different footnotes, like this:

![tables](https://cloud.githubusercontent.com/assets/1944744/14317835/16e63f94-fbd8-11e5-91f0-37bba5610ea5.jpg)

Is this accessible? If not, how should we approach it?

**Comment by rlambert27** [comment]
Personally I've always felt like WCAG isn't clear about the use of the asterisk only. Considering F26 is a failure for only using a graphical symbol to indicate information I usually fail all pages that use the asterisk without also providing an explanation of its use. For example a form may use the asterisk to indicate required form fields as long as the top also has "The asterisk (*) symbol in the following form indicates required fields.".

In this case here unless you explain, at the top of the page, the meaning of the asterisk symbol I would say this would not be accessible. I'm also assuming in this case that there is no hidden link text for those links which would be very confusing for a visually impaired user trying to find a specific footnote on the page.

Keep in mind as well that the default verbosity level set by JAWS will not read the asterisk symbol. However in this case since it's a link and it's the only item in the link I'm not sure how JAWS will react. If you have the URL to the page above we could test that for you.

My suggestion would be to switch over to using numbers like in the footnotes plugin (http://wet-boew.github.io/v4.0-ci/demos/footnotes/footnotes-en.html). You can even go a little further with your hidden text and indicate the footnotes is for which table.

Hope that helps!

**Comment by bsouster** [comment]
Actually, we do have hidden text to label it better for screen readers. I suppose that does make it unique link texts for visually-impaired users if they're read as I'd hope.

Here's the [example above](http://www.cic.gc.ca/english/resources/evaluation/cec/2016/program.asp).

**Comment by bsouster** [comment]
@rlambert27 I think you misunderstood a bit what I was asking. We're using the asterisk example in the WET footer plugin example page.

The issue I'm referring to is when multiple tables in the same page using asterisks. It causes multiple links with the same link text going to different destinations (see example in my first post).

You reminded me of the hidden link text, so I'm thinking adding more context to the hidden text will give it the detail it needs. (Footnote * -> Table 5.2 footnote *)

Example:
`<h4` id="fn-t5-1" class="wb-inv">Table 5.2 footnotes</h4>
`	<dl>
`    <dt>Table 5.2 footnote *</dt>
`		<dd id="fn-t5-1*">
`			<p>Total program costs take into account various costs associated with program delivery, program management and departmental support processes internal and external to CIC. They do not, however, represent the full Government of Canada costs, as they do not include all CIC and other government department (OGD) costs.</p>
`			<p class="fn-rtn">
`				<a href="#fn-t5-1*-rf"><span class="wb-inv">Return to table 5.2 footnote </span>*<span class="wb-inv"> referrer</span></a>
`			</p>
`		</dd>
`    <dt>Table 5.2 footnote **</dt>
`		<dd id="fn-t5-1**">
`			<p>The number of applications received for FY 2010-11 may be inconsistent due to the transition of application processing from the visa office in Buffalo to Centralized Processing Centre (CPC) Ottawa.</p>
`			<p class="fn-rtn">
`				<a href="#fn-t5-1**-rf"><span class="wb-inv">Return to Table 5.2 footnote </span>**<span class="wb-inv"> referrer</span></a>
`			</p>
`		</dd>
`	</dl>

**Comment by rlambert27** [comment]
A quick test shows that the link will read "Samepage link. Click on star” and “Same page link click on star star”. The return URLs as suggested look good.

I would change the use of the asterisk or make the URLs unique. For example currently it's:

```
<a class="fn-lnk" href="#fn-t5-1**"><span class="wb-inv">Footnote </span>**</a>
```

i would use:

```
<a class="fn-lnk" href="#fn-t5-1**"><span class="wb-inv">Footnote for table 5.2 </span>**</a>
```

Anyway hope that helps!

**Comment by MaximMartel** [comment]
Is this issue resolved by the proposed solution of @rlambert27 ? @bsouster

**Comment by bsouster** [comment]
This issue is resolved.
